packages/libs/lambda-at-edge/src/default-handler.ts

```diff
@@ -247,6 +247,13 @@ const handleOriginResponse = async ({
       return response;
     }
 
+    // Set 500 status code for 500.html page. We do not need normalised URI as it will always be "/404.html"
+    if (s3Uri.endsWith("/500.html")) {
```

Comment:
Will add a test for this. In the future if/when origin response handler is removed, it could be moved to CloudFront error page or we could use aws-sdk s3-client to retrieve the page from S3 while in the origin request handler (and thus be able to set the status code there).